### ISSUE TITLE
Cart dispenser outputting its own redstone signal

### ISSUE THREAD

**Issue by minecraftothersteve** [railcraft-reborn/railcraft#258]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Bug

I've been messing around with this mod and think I've found a bug with the cart dispenser.
When I give it a redstone signal, it'll pick up or place a cart as expected, but when the redstone signal from the source turns off the dispenser seems to be creating it's own signal and kinda locks up the system.
A couple screenshots to show what I mean https://imgur.com/a/Ja8oVAn
The first screenshot is after I turn on the lever, and you can see the redstone signal going into the dispenser is 14. Second is after I turn off the lever, where the signal is now coming out of the dispenser at a 15. It does this even if there are no carts in the system. 
Turning the lever on again has no effect on the dispenser, since the redstone signal is already high so cannot receive a new pulse.

### Railcraft Reborn version

1.16

### NeoForge/Forge version

Forge 47.2.14

### Minecraft version

1.20.1

### Minecraft server

No

### To Reproduce

1. Place cart dispenser, 1+ redstone dust and a lever in a row
2. Turn on the lever
3. Turn off the lever

### Expected behavior

When the lever is turned off, the redstone signal should go low to 0. The cart dispenser should not be generating its own signal.

### Screenshots & Video

Screenshots https://imgur.com/a/Ja8oVAn

**Comment by minecraftothersteve** [comment]
Doing some more testing with a locking track next to the dispenser, since that's how I first discovered the issue. Here are some screenshots that show a step-by-step test, showing signs of phantom redstone signals that might be caused by a block update issue https://imgur.com/a/FIR8dA0

One solution I've found is to place a repeater next to the dispenser to push the redstone signal into it. Using this method seems to be working with the locking track setup so that the dispenser and locking track are only powered when the lever is turned on, and both are unpowered when the lever is off.
